### ISSUE TITLE
No stacktrace on non-fatal event. "Discarded invalid value for parameter 'threads'"

### ISSUE THREAD

**Issue by edenman** [getsentry/sentry-cocoa#176]
_Platform: iOS
_Swift: 3.0
_sentry-cocoa installed with: CocoaPods

Version of sentry-cocoa:
**3.0.1**

---
I have following issue:

I log a non-fatal event to Sentry thusly:
```
func LogNonFatal(_ message: String) {
	let sentry: Client = Client.shared!
	let isMainThead = Thread.isMainThread
	sentry.snapshotStacktrace {
		let event = Event(level: .error)
		event.message = message
		if event.extra == nil {
			event.extra = [:]
		}
		event.extra!["is-main-thread"] = isMainThead
		sentry.send(event: event)
	}
}
```
There's no stacktrace on the resulting sentry event: https://sentry.io/recharge/consumer-ios/issues/269651474/ 

There's an error on the Sentry web UI: "Discarded invalid value for parameter 'threads' " and then it has the whole threads json included if you expand the error.

**Comment by HazAT** [comment]
Hey, this was due to a bug that a frame was empty, this has been fixed in `3.0.4`.
Please update to the latest version (`3.0.9`) and try again, this should work already.

**Comment by edenman** [comment]
k.  @HazAT is there a changelog somewhere for this library?  I don't see a `CHANGELOG.md` in the root dir of this repo.

**Comment by HazAT** [comment]
No `CHANGELOG.md`, this is pretty much the changelog https://github.com/getsentry/sentry-cocoa/releases

**Comment by edenman** [comment]
This has started happening to me again:
https://sentry.io/recharge/consumer-ios/issues/311017168/
https://sentry.io/recharge/consumer-ios/issues/311021593/

I think it's probably a race condition?  ie if I call my `LogNonFatal` method a few times in a row, some of them fail to get a stacktrace but still upload an issue.

**Comment by HazAT** [comment]
Hey, I am currently looking into this, there is definitely a bug in the client.
I will keep you updated.

**Comment by HazAT** [comment]
So I've just released version `3.2.0` which fixes the issue in the tests I did.
I've experienced this issue especially when calling it multiple times.
In order to make it work reliably you have to add one function call ...
```
func LogNonFatal(_ message: String) {
	let sentry: Client = Client.shared!
	let isMainThead = Thread.isMainThread
	sentry.snapshotStacktrace {
		let event = Event(level: .error)
		event.message = message
		if event.extra == nil {
			event.extra = [:]
		}
		event.extra!["is-main-thread"] = isMainThead


                // add stored stacktrace to event
                sentry.appendStacktrace(to: event)



		sentry.send(event: event)
	}
}
```
